node/impl/full/trace.go

```diff
@@ -0,0 +1,263 @@
+package full
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/stmgr"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+)
+
+type EthTraceAPI interface {
+	TraceBlock(ctx context.Context, blkNum string) (interface{}, error)
+	TraceReplayBlockTransactions(ctx context.Context, blkNum string, traceTypes []string) (interface{}, error)
+}
+
+var (
+	_ EthTraceAPI = *new(api.FullNode)
+)
+
+type EthTrace struct {
```

Comment:
This should use `fx.In`, that would make it not need a constructor (and be consistent with other API impl structs)